### PR TITLE
UHM-3461: Add patient Header

### PR DESCRIPTION
**Pull request by jomadoye**
screen shot 2018-09-04 at 2 44 36 pm

### CODE REVIEW

File: app/js/components/App.jsx [hunk 1]
```diff
@@ -7,15 +7,43 @@
...
+import { PatientHeader } from '@openmrs/react-components';
 
+import patientAction from '../actions/patientAction';
 import LabOrdersList from './LabOrdersList';
 
-export default class App extends React.Component {
+export class App extends React.Component {
+  componentDidMount() {
+    const { dispatch } = this.props;
+    dispatch(patientAction.getPatientRecord('da5e210f-a3c4-4c49-80f2-e2e5386db8ad'));
```
Comment (jomadoye) [inline app/js/components/App.jsx]: I am hardcoding the Patient ID for now, till we can get it from the URL

File: app/js/actions/actionTypes.js [hunk 2]
```diff
@@ -1,2 +1,14 @@
...
+  'SUCCEEDED',
+  'FAILED',
+  'REQUESTED',
+];
+
 export const FETCH_LAB_ORDERS = 'FETCH_LAB_ORDERS';
+
+export default {
+  SET_PATIENT: types('set_patient', BASIC_TYPES),
+  SET_PATIENT_NOTE: types('set_patient_note', BASIC_TYPES),
```
Comment (mogoodrich) [inline app/js/actions/actionTypes.js]: Let's just remove "SET_PATIENT_NOTE" for now... it was something added to the header, but we never use it...

File: app/js/actions/patientAction.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+});
+
+const getPatientRecordFailed = message => ({
+  type: actionTypes.SET_PATIENT.FAILED,
+  error: {
+    message,
+  },
+});
+
+const getPatientNote = patientUuid => ({
```
Comment (mogoodrich) [inline app/js/actions/patientAction.js]: Let's just remove these three note actions

File: app/js/components/App.jsx [hunk 4]
```diff
@@ -7,15 +7,43 @@
...
+    dispatch(patientAction.getPatientNote('da5e210f-a3c4-4c49-80f2-e2e5386db8ad'));
+  }
+
   render() {
+    const { patientHeaderDetail, patientNote } = this.props;
     return (
       <div>
+        {/* TODO: Work on conditionally rendering the patient header based on the route */}
+        <PatientHeader patient={patientHeaderDetail} note={patientNote} />
```
Comment (mogoodrich) [inline app/js/components/App.jsx]: Just remove the patient note here as well.

File: app/js/actions/patientAction.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import actionTypes from "./actionTypes";
+
+const getPatientRecord = patientUuid => ({
```
Comment (mogoodrich) [inline app/js/actions/patientAction.js]: Per our conventions, we can just call this "gePatient", we don't need to call it "getPatientRecord"

Same with "getPatientRecordSucceeded" and "getPatientRecordFailed"

File: app/js/rest/patientHeaderRest.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import axiosInstance from '../config/axios';
+
+const api = {
+  getPatientRecord: params => axiosInstance.get(`patient/${params.patientUuid}?v=custom:(patientId,uuid,patientIdentifier:(uuid,identifier),person:(gender,age,birthdate,birthdateEstimated,personName,preferredAddress),attributes:(value,attributeType:(name)))`)
```
Comment (mogoodrich) [inline app/js/rest/patientHeaderRest.js]: Let's rename this getPatient and actually move into patientRest.js in react-components:

https://github.com/openmrs/openmrs-react-components/blob/master/src/rest/patientRest.js

File: app/js/components/App.jsx (hunk 1 above)
Comment (mogoodrich) [reply app/js/components/App.jsx]: Cool, this is an existing patient on HUM-CI?

**Comment by mogoodrich** [comment]
Looks good now, let's go ahead and merge this?

**Pull request merged**